### PR TITLE
feat: add blogs CRUD

### LINKED ISSUE #6: Work on creating Blog Endpoints
Blocker: #3 

Create CRUD API routes for blogs

### PR DESCRIPTION
**Pull request by Anupamaraie**
## Added blogs app

Description:
Contains all the functionality for the blogs page in the form of an API. Blog endpoints, in general.

Closes #6

### CODE REVIEW

**Comment by Br9ha20** [comment]
@thenishantsapkota Is this reviewed? If so let's merge it.

File: blogs/serializers.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+from django.contrib.auth.models import User
+from rest_framework import serializers
+
+from authentication.serializers import *
+
+from .models import *
```
Comment (thenishantsapkota) [inline blogs/serializers.py]: Don't use wildcard imports here.

File: blogs/serializers.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+from django.contrib.auth.models import User
+from rest_framework import serializers
+
+from authentication.serializers import *
```
Comment (thenishantsapkota) [inline blogs/serializers.py]: Why are we using wildcard imports?
Please fix this

File: blogs/urls.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    path("createblog", CreateBlogView.as_view()),
+    path("createblog/<int:post_id>/", CreateBlogView.as_view()),
+    path("deleteblog/<int:post_id>", CreateBlogView.as_view()),
+    path("createcomment", CreateCommentView.as_view()),
+    path("readcomment/<int:id>", ReadCommentView.as_view()),
+    path("editcomment/<int:id>/", EditCommentView.as_view()),
+    path("deletecomment/<int:id>", CreateCommentView.as_view()),
+    path("likepost/<int:post_id>", LikeCreateView.as_view()),
+    path("unlikepost/<int:post_id>", LikeCreateView.as_view()),
+]
```
Comment (thenishantsapkota) [inline blogs/urls.py]: Always append a `/` at the end of the urls please

File: blogs/views.py [hunk 4]
```diff
@@ -1,3 +1,265 @@
+from django.http import JsonResponse
```
Comment (thenishantsapkota) [inline blogs/views.py]: We're not using django's JsonResponse

File: blogs/views.py [hunk 5]
```diff
@@ -1,3 +1,265 @@
...
+            Response: The HTTP response object.
+        """
+
+        posts = Posts.objects.filter(post_id=post_id).first()
+        if not posts:
+            return cr.error(message="Post not found.")
+        serializer = self.serializer_class(posts)
+        return cr.success(data=serializer.data, message="Blog fetched successfully!")
+
+
```
Comment (thenishantsapkota) [inline blogs/views.py]: This view is useless, we could have used the `GetBlogView` for this implementation as well!

File: blogs/views.py [hunk 6]
```diff
@@ -1,3 +1,265 @@
...
+        """
+
+        comment = Comments.objects.filter(id=id).first()
+        if not comment:
+            return cr.error(message="Comment not found.")
+        serializer = self.serializer_class(comment)
+        return cr.success(data=serializer.data, message="Comment fetched successfully!")
+
+
+class LikeCreateView(APIView):
```
Comment (thenishantsapkota) [inline blogs/views.py]:
```suggestion
class LikeView(APIView):
```

**Review by thenishantsapkota — CHANGES_REQUESTED** [review]
Please make the requested changes and push the changes. Make sure to test the existing features well.

File: blogs/serializers.py (hunk 1 above)
Comment (Anupamaraie) [reply blogs/serializers.py]: okii

File: blogs/serializers.py (hunk 2 above)
Comment (Anupamaraie) [reply blogs/serializers.py]: idk what that does so i removed it.

File: blogs/urls.py (hunk 3 above)
Comment (Anupamaraie) [reply blogs/urls.py]: okii

File: blogs/views.py (hunk 5 above)
Comment (Anupamaraie) [reply blogs/views.py]: I thought we couldn't use two get functions in the same class. lol

**Comment by thenishantsapkota** [comment]
Please restore the `.env.example` file

File: blogs/admin.py [hunk 7]
```diff
@@ -1,3 +1,8 @@
 from django.contrib import admin
 
+from .models import *
```
Comment (thenishantsapkota) [inline blogs/admin.py]: Same here remove wild card imports please

File: blogs/models.py [hunk 8]
```diff
@@ -1,3 +1,45 @@
+from datetime import datetime
+
+from django.conf import settings
+from django.contrib.auth.models import User
```
Comment (thenishantsapkota) [inline blogs/models.py:2]: Unused import here.

**Review by thenishantsapkota — CHANGES_REQUESTED** [review]
Make the `Like` api have only PUT method.

If a user sends a request to like endpoint, try to fetch the like for a user in a post. If like exists delete the like esle create a new like.
Creating two seperate api endpoints for liking doesn't make sense and will make it harder for front end developers to implement it.

**Review by thenishantsapkota — COMMENTED** [review]
Code looks good to me.
I'll test them tommorow and approve them if they're working fine!

**Comment by Br9ha20** [comment]
@thenishantsapkota Is this tested and approved?

**Pull request merged**